### PR TITLE
fontbakery: init at 0.10.40

### PR DESCRIPTION
**Pull request by danc86**
###### Description of changes

[Font Bakery](https://github.com/googlefonts/fontbakery) is a command-line tool for checking the quality of font projects.

It's used for example in the build process for the Inter font family.

This PR also includes a number of new Python packages which are dependencies of fontbakery.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- ~[ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~
- [x] Tested, as applicable:
  - ~[NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - ~or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)~
  - ~made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages~
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - ~[ ] (Package updates) Added a release notes entry if the change is major or breaking~
  - ~[ ] (Module updates) Added a release notes entry if the change is significant~
  - ~[ ] (Module addition) Added a release notes entry if adding a new NixOS module~
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by danc86** [comment]
Result of `nixpkgs-review pr 228099` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>85 packages built:</summary>
  <ul>
    <li>dehinter (python310Packages.dehinter)</li>
    <li>dehinter.dist (python310Packages.dehinter.dist)</li>
    <li>font-v (python310Packages.font-v)</li>
    <li>font-v.dist (python310Packages.font-v.dist)</li>
    <li>fontbakery (python310Packages.fontbakery)</li>
    <li>fontbakery.dist (python310Packages.fontbakery.dist)</li>
    <li>opentype-sanitizer</li>
    <li>python310Packages.axisregistry</li>
    <li>python310Packages.axisregistry.dist</li>
    <li>python310Packages.babelfont</li>
    <li>python310Packages.babelfont.dist</li>
    <li>python310Packages.beziers</li>
    <li>python310Packages.beziers.dist</li>
    <li>python310Packages.collidoscope</li>
    <li>python310Packages.collidoscope.dist</li>
    <li>python310Packages.commandlines</li>
    <li>python310Packages.commandlines.dist</li>
    <li>python310Packages.fontfeatures</li>
    <li>python310Packages.fontfeatures.dist</li>
    <li>python310Packages.gflanguages</li>
    <li>python310Packages.gflanguages.dist</li>
    <li>python310Packages.glyphsets</li>
    <li>python310Packages.glyphsets.dist</li>
    <li>python310Packages.glyphtools</li>
    <li>python310Packages.glyphtools.dist</li>
    <li>python310Packages.kurbopy</li>
    <li>python310Packages.kurbopy.dist</li>
    <li>python310Packages.opentypespec</li>
    <li>python310Packages.opentypespec.dist</li>
    <li>python310Packages.ots-python</li>
    <li>python310Packages.ots-python.dist</li>
    <li>python310Packages.rstr</li>
    <li>python310Packages.rstr.dist</li>
    <li>python310Packages.sre-yield</li>
    <li>python310Packages.sre-yield.dist</li>
    <li>python310Packages.stringbrewer</li>
    <li>python310Packages.stringbrewer.dist</li>
    <li>ufolint (python310Packages.ufolint)</li>
    <li>ufolint.dist (python310Packages.ufolint.dist)</li>
    <li>python310Packages.vharfbuzz</li>
    <li>python310Packages.vharfbuzz.dist</li>
    <li>python310Packages.youseedee</li>
    <li>python310Packages.youseedee.dist</li>
    <li>python311Packages.axisregistry</li>
    <li>python311Packages.axisregistry.dist</li>
    <li>python311Packages.babelfont</li>
    <li>python311Packages.babelfont.dist</li>
    <li>python311Packages.beziers</li>
    <li>python311Packages.beziers.dist</li>
    <li>python311Packages.collidoscope</li>
    <li>python311Packages.collidoscope.dist</li>
    <li>python311Packages.commandlines</li>
    <li>python311Packages.commandlines.dist</li>
    <li>python311Packages.dehinter</li>
    <li>python311Packages.dehinter.dist</li>
    <li>python311Packages.font-v</li>
    <li>python311Packages.font-v.dist</li>
    <li>python311Packages.fontbakery</li>
    <li>python311Packages.fontbakery.dist</li>
    <li>python311Packages.fontfeatures</li>
    <li>python311Packages.fontfeatures.dist</li>
    <li>python311Packages.gflanguages</li>
    <li>python311Packages.gflanguages.dist</li>
    <li>python311Packages.glyphsets</li>
    <li>python311Packages.glyphsets.dist</li>
    <li>python311Packages.glyphtools</li>
    <li>python311Packages.glyphtools.dist</li>
    <li>python311Packages.kurbopy</li>
    <li>python311Packages.kurbopy.dist</li>
    <li>python311Packages.opentypespec</li>
    <li>python311Packages.opentypespec.dist</li>
    <li>python311Packages.ots-python</li>
    <li>python311Packages.ots-python.dist</li>
    <li>python311Packages.rstr</li>
    <li>python311Packages.rstr.dist</li>
    <li>python311Packages.sre-yield</li>
    <li>python311Packages.sre-yield.dist</li>
    <li>python311Packages.stringbrewer</li>
    <li>python311Packages.stringbrewer.dist</li>
    <li>python311Packages.ufolint</li>
    <li>python311Packages.ufolint.dist</li>
    <li>python311Packages.vharfbuzz</li>
    <li>python311Packages.vharfbuzz.dist</li>
    <li>python311Packages.youseedee</li>
    <li>python311Packages.youseedee.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/youseedee/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/youseedee/default.nix:18]: 1. Add a comment near the patch, and squash this commit to the 
2. Instead of the `substituteInPlace` used in `preBuild`, use the Nix function `substituteAll`.
3. Squash this commit to the previous, `init` commit.

File: pkgs/development/python-modules/glyphtools/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/glyphtools/default.nix:22]: Nit: Usually in Python's `.pkgs`, explicitly enable or disable tests with `doCheck`. Could you verify that indeed tests are being run here (with the default `doCheck = true` value)?

File: pkgs/development/python-modules/youseedee/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/youseedee/default.nix]: Same here, `doCheck = false` won't hurt.

File: pkgs/development/python-modules/youseedee/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/youseedee/default.nix:32]: So use `doCheck = true` :)

File: pkgs/development/python-modules/fontfeatures/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/fontfeatures/default.nix:56]: Is it possible to use `disabledTests` here instead?

File: pkgs/development/python-modules/babelfont/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/babelfont/default.nix:24]: I think 3.0.1 is tagged now.

File: pkgs/development/python-modules/kurbopy/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/kurbopy/default.nix:38]: Upstream tracks `Cargo.lock` file now, so these are no longer needed.

File: pkgs/tools/security/opentype-sanitizer/default.nix
Comment (doronbehar) [inline pkgs/tools/security/opentype-sanitizer/default.nix:13]: The modern approach nowadays is:

1. Use `pkgs/by-name/`.
2. Use `stdenv.mkDerivation(finalAttrs: {`...

File: pkgs/development/python-modules/ots-python/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/ots-python/default.nix:20]: Similar to a different comment, use `substituteAll` nix function, and comment something near the patch.

File: pkgs/development/python-modules/stringbrewer/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/stringbrewer/default.nix:27]: That's how you do it :) :+1:.

File: pkgs/development/python-modules/vharfbuzz/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/vharfbuzz/default.nix:11]: I assume this downgrade decision might need to be revisited now - maybe upstream improved the situation. If not, I'd squash this commit and it's parent into a single commit like this:

```
python3Packages.vharfbuzz0_1_4: init at 0.1.4
```

Then, put the information answering the question "why isn't the latest and greatest version of this project packaged?", that is available in the commit message, near the version string.

File: pkgs/development/python-modules/axisregistry/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/axisregistry/default.nix:32]: This can also be squashed to the `axisregistry: init` commit. Also, put the reason explained in the commit message in a comment, instead of in the commit message.

File: pkgs/development/python-modules/gflanguages/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/gflanguages/default.nix:34]: Same here.

File: pkgs/top-level/all-packages.nix
Comment (doronbehar) [inline pkgs/top-level/all-packages.nix:1511]: Is this package supposed to be available as a python module as well?

File: pkgs/development/python-modules/fontbakery/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/fontbakery/default.nix:131]: Nit: It might be safer to use `installShellFiles`.

File: pkgs/development/python-modules/fontbakery/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/fontbakery/default.nix:61]: Tests that have open PRs, please use `fetchpatch`. I guess many of these patches are already merged upstream so hopefully this will look different after you'd revisit this PR.

**Review by doronbehar — CHANGES_REQUESTED** [review]
Great PR overall! General tip: Because there are so many new packages, it'd be nice to test them on Darwin with ofborg as well. You could trigger that automatically if all the commits' messages' titles will be change from `python3Packages...` to `python311Packages...`. Only the latest is directly evaluated by Hydra, and ofborg counts on that as well.

Also, sorry for waiting so long for a review. We'll sure get all of these packages as quick as possible.

P.S There's a small, trivial merge conflict.

File: pkgs/top-level/all-packages.nix
Comment (jopejoe1) [reply pkgs/top-level/all-packages.nix:1511]: It required as a module for https://github.com/notofonts/notobuilder/ which we would need if we ever want to build noto-fonts from source

**Comment by danc86** [comment]
Thanks for the review, and for the tip about `python310Packages` in the commit messages.

I'll rebase and update this PR in the next few days when I get some spare time.

File: pkgs/development/python-modules/glyphtools/default.nix
Comment (danc86) [reply pkgs/development/python-modules/glyphtools/default.nix:22]: Yes, I got the tests passing on all these packages (except ones where I couldn't find any reasonable test suite to run). I'll add an explicit `doCheck = true` to make it clear there are tests.

File: pkgs/development/python-modules/fontfeatures/default.nix
Comment (danc86) [reply pkgs/development/python-modules/fontfeatures/default.nix:56]: No, that didn't work because it's a parametrized test... but the test is passing in the latest upstream release now.

File: pkgs/development/python-modules/vharfbuzz/default.nix
Comment (danc86) [reply pkgs/development/python-modules/vharfbuzz/default.nix:11]: Actually it was nixpkgs not upstream that was to blame here :-) our uharfbuzz was too old, and I was planning to send a PR at some point to upgrade it... of course in the meantime it has already been upgraded, so this indeed no longer an issue.

File: pkgs/top-level/all-packages.nix
Comment (danc86) [reply pkgs/top-level/all-packages.nix:1511]: All the packages in this PR should be available as Python modules (since a lot of font build processes will depend on them that way, even if they're really a command line tool like fontbakery). Plus for all those packages where upstream's docs give some indication that the tool is designed to be invoked from the command line, I've exported them as Python applications as well.

**Comment by danc86** [comment]
I've rebased this PR, fixed the conflicts, and amended it as follows:

* latest upstream release of all packages
* added shaperglot, new dependency of fontbakery
* squashed all changes into each "init" commit
* used explicit `doCheck = true`
* used `substituteAll` nix function on nixpkgs-specific patches and commented them
* switched to upstream `Cargo.lock` for kurbopy
* moved opentype-sanitizer under `pkgs/by-name` and removed `rec` in favour of `mkDerivation (finalAttrs:`
* used `fetchpatch` for the one remaining open PR on fontbakery
* used `python311Packages` in all the commit messages, to trigger ofborg properly

**Comment by danc86** [comment]
Result of `nixpkgs-review pr 228099` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>89 packages built:</summary>
  <ul>
    <li>dehinter (python311Packages.dehinter)</li>
    <li>dehinter.dist (python311Packages.dehinter.dist)</li>
    <li>font-v (python311Packages.font-v)</li>
    <li>font-v.dist (python311Packages.font-v.dist)</li>
    <li>fontbakery (python311Packages.fontbakery)</li>
    <li>fontbakery.dist (python311Packages.fontbakery.dist)</li>
    <li>opentype-sanitizer</li>
    <li>python310Packages.axisregistry</li>
    <li>python310Packages.axisregistry.dist</li>
    <li>python310Packages.babelfont</li>
    <li>python310Packages.babelfont.dist</li>
    <li>python310Packages.beziers</li>
    <li>python310Packages.beziers.dist</li>
    <li>python310Packages.collidoscope</li>
    <li>python310Packages.collidoscope.dist</li>
    <li>python310Packages.commandlines</li>
    <li>python310Packages.commandlines.dist</li>
    <li>python310Packages.dehinter</li>
    <li>python310Packages.dehinter.dist</li>
    <li>python310Packages.font-v</li>
    <li>python310Packages.font-v.dist</li>
    <li>python310Packages.fontbakery</li>
    <li>python310Packages.fontbakery.dist</li>
    <li>python310Packages.fontfeatures</li>
    <li>python310Packages.fontfeatures.dist</li>
    <li>python310Packages.gflanguages</li>
    <li>python310Packages.gflanguages.dist</li>
    <li>python310Packages.glyphsets</li>
    <li>python310Packages.glyphsets.dist</li>
    <li>python310Packages.glyphtools</li>
    <li>python310Packages.glyphtools.dist</li>
    <li>python310Packages.kurbopy</li>
    <li>python310Packages.kurbopy.dist</li>
    <li>python310Packages.opentypespec</li>
    <li>python310Packages.opentypespec.dist</li>
    <li>python310Packages.ots-python</li>
    <li>python310Packages.ots-python.dist</li>
    <li>python310Packages.rstr</li>
    <li>python310Packages.rstr.dist</li>
    <li>python310Packages.shaperglot</li>
    <li>python310Packages.shaperglot.dist</li>
    <li>python310Packages.sre-yield</li>
    <li>python310Packages.sre-yield.dist</li>
    <li>python310Packages.stringbrewer</li>
    <li>python310Packages.stringbrewer.dist</li>
    <li>python310Packages.ufolint</li>
    <li>python310Packages.ufolint.dist</li>
    <li>python310Packages.vharfbuzz</li>
    <li>python310Packages.vharfbuzz.dist</li>
    <li>python310Packages.youseedee</li>
    <li>python310Packages.youseedee.dist</li>
    <li>python311Packages.axisregistry</li>
    <li>python311Packages.axisregistry.dist</li>
    <li>python311Packages.babelfont</li>
    <li>python311Packages.babelfont.dist</li>
    <li>python311Packages.beziers</li>
    <li>python311Packages.beziers.dist</li>
    <li>python311Packages.collidoscope</li>
    <li>python311Packages.collidoscope.dist</li>
    <li>python311Packages.commandlines</li>
    <li>python311Packages.commandlines.dist</li>
    <li>python311Packages.fontfeatures</li>
    <li>python311Packages.fontfeatures.dist</li>
    <li>python311Packages.gflanguages</li>
    <li>python311Packages.gflanguages.dist</li>
    <li>python311Packages.glyphsets</li>
    <li>python311Packages.glyphsets.dist</li>
    <li>python311Packages.glyphtools</li>
    <li>python311Packages.glyphtools.dist</li>
    <li>python311Packages.kurbopy</li>
    <li>python311Packages.kurbopy.dist</li>
    <li>python311Packages.opentypespec</li>
    <li>python311Packages.opentypespec.dist</li>
    <li>python311Packages.ots-python</li>
    <li>python311Packages.ots-python.dist</li>
    <li>python311Packages.rstr</li>
    <li>python311Packages.rstr.dist</li>
    <li>shaperglot (python311Packages.shaperglot)</li>
    <li>shaperglot.dist (python311Packages.shaperglot.dist)</li>
    <li>python311Packages.sre-yield</li>
    <li>python311Packages.sre-yield.dist</li>
    <li>python311Packages.stringbrewer</li>
    <li>python311Packages.stringbrewer.dist</li>
    <li>ufolint (python311Packages.ufolint)</li>
    <li>ufolint.dist (python311Packages.ufolint.dist)</li>
    <li>python311Packages.vharfbuzz</li>
    <li>python311Packages.vharfbuzz.dist</li>
    <li>python311Packages.youseedee</li>
    <li>python311Packages.youseedee.dist</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
Sorry for not responding to this. I haven't forgotten this PR. Hopefully will find time to approve it tomorrow or next week.

**Review by jopejoe1 — APPROVED** [review]

File: pkgs/development/python-modules/glyphsets/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/glyphsets/default.nix:25]: Didn't `pythonRelaxDepsHook` work? If so, explain why or at least mention this in the comment as well. See also this:

https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574/40

File: pkgs/development/python-modules/babelfont/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/babelfont/default.nix:28]: Nowadays people use `pyproject = true;` instead. See:

https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters

File: pkgs/development/python-modules/kurbopy/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/kurbopy/default.nix:40]: Hmm interesting. Could you explain this a bit in a comment?

File: pkgs/development/python-modules/shaperglot/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/shaperglot/default.nix:28]: Again I think this should be switched to `pyproject = true;`.

**Review by doronbehar — COMMENTED** [review]
Very well written! Thanks for this big contribution, and again sorry for this one more iteration. I will have time tomorrow for sure to merge this if these small issues will be fixed!

**Comment by doronbehar** [comment]
Also, there's a merge conflict, now I'm noticing - probably fixable easily.

File: pkgs/development/python-modules/glyphsets/default.nix
Comment (danc86) [reply pkgs/development/python-modules/glyphsets/default.nix:25]: `pythonRelaxDepsHook` doesn't work here because that works by manipulating the wheel after it's built, whereas the problematic requirement here is in `setup_requires` which means it fails before it can get to that point:

```
Traceback (most recent call last):
  File "/build/glyphsets-0.6.5/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 26, in <module>
    setup(
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/__init__.py", line 102, in setup
    _install_setup_requires(attrs)
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/__init__.py", line 75, in _install_setup_requires
    _fetch_build_eggs(dist)
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/__init__.py", line 80, in _fetch_build_eggs
    dist.fetch_build_eggs(dist.setup_requires)
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/dist.py", line 662, in fetch_build_eggs
    return _fetch_build_eggs(self, requires)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/installer.py", line 38, in _fetch_build_eggs
    resolved_dists = pkg_resources.working_set.resolve(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/pkg_resources/__init__.py", line 865, in _resolve_dist
    dist = best[req.key] = env.best_match(
                           ^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1135, in best_match
    return self.obtain(req, installer)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1147, in obtain
    return installer(requirement)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/y8yq448xka665bqyd5gvdhajqdbsnyys-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/installer.py", line 103, in _fetch_build_egg_no_warn
    raise DistutilsError(str(e)) from e
distutils.errors.DistutilsError: Command '['/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/bin/python3.11', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmpf6na7nbg', '--quiet', 'setuptools_scm<6.1,>=4']' returned non-zero exit status 1.
```

I'll mention this in the patch comment.

File: pkgs/development/python-modules/kurbopy/default.nix
Comment (danc86) [reply pkgs/development/python-modules/kurbopy/default.nix:40]: This is a bit of a dodgy hack and I shouldn't have left it in (not without a comment).

The actual problem is the classic pytest import confusion. Because of pytest's manipulations of sys.path, the test process ends up importing `kurbopy/__init__.py` from the source directory, which does not contain the built .so, and so it fails like this:

```
/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_basic.py:1: in <module>
    from kurbopy import Point, Vec2, TranslateScale
kurbopy/__init__.py:1: in <module>
    from .kurbopy import Point as Point
E   ModuleNotFoundError: No module named 'kurbopy.kurbopy'
```

We want it to import kurbopy from the nix store via PYTHONPATH instead.

So this hack was stealing `$dist` and using it to populate the source directory with a usable copy of the built .so, but a proper solution would be to make pytest import kurbopy from the installed directory like a proper installCheckPhase is supposed to do.

Pytest has an `--import-mode` option which should help with that kind of mess. But no matter what I do, `sys.path` always imports from `.` first. I think it's because `pytestCheckHook` does `python -m pytest`, and that's one of the side effects [noted in the docs](https://docs.pytest.org/en/7.1.x/how-to/usage.html#calling-pytest-through-python-m-pytest).

Anyway, I can just hack it to delete the module from the source directory in `preCheck`, so that it imports the installed copy from the nix store instead. With a comment explaining what's going on. :-)

**Comment by danc86** [comment]
I've rebased this and amended it as follows:

* `format = "pyproject"` -> `pyproject = true`
* added a comment to glyphsets explaining why `pythonRelaxDepsHook` doesn't work
* handle pytest import path problems in kurbopy in a cleaner way, with an explanatory comment

File: pkgs/development/python-modules/kurbopy/default.nix
Comment (doronbehar) [reply pkgs/development/python-modules/kurbopy/default.nix]: > Anyway, I can just hack it to delete the module from the source directory in `preCheck`, so that it imports the installed copy from the nix store instead. With a comment explaining what's going on. :-)

Thanks for the thorough explanation @danc86 . Could you do me please 1 more favor? In the very well written comment, could you mention that this is a common issue in nixpkgs and it is investigated in https://github.com/NixOS/nixpkgs/issues/255262 ?

Comment (doronbehar) [reply pkgs/development/python-modules/kurbopy/default.nix]: The PR should be ready after issue https://github.com/NixOS/nixpkgs/issues/255262 is mentioned.

File: pkgs/development/python-modules/ots-python/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/ots-python/default.nix:11]: 1 last small thing please @danc86 : Could you make both the attribute name of this package, `pname` and the file name all be `opentype-sanitizer`? This would also allow you to `inherit pname version;` in `fetchPypi`. I didn't notice this issue in my previous review, and I remember Python ecosystem maintainers are rather strict regarding this habit.

File: pkgs/development/python-modules/ots-python/default.nix
Comment (doronbehar) [inline pkgs/development/python-modules/ots-python/default.nix:44]: The homepage naturally will stay the same.

File: pkgs/development/python-modules/ots-python/default.nix
Comment (doronbehar) [reply pkgs/development/python-modules/ots-python/default.nix:11]: You are right, I just noticed that now and I was about to comment. It may cause issues, and the current situation is not so bad.

**Review by doronbehar — APPROVED** [review]

**Comment by danc86** [comment]
Thanks for the reviews. :-)

**Pull request merged**
